src/langcheck/metrics/metric_inputs.py

```diff
@@ -0,0 +1,454 @@
+from __future__ import annotations
+
+from typing import List, Union
+
+import pandas as pd
+from jinja2 import Environment, meta
+
+SingleInputType = Union[str, List[str], None]
+
+
+def _map_pairwise_input_to_list(
+    input: tuple[SingleInputType, SingleInputType],
+) -> tuple[list[str] | None, list[str] | None]:
+    return (
+        _map_single_input_to_list(input[0]),
+        _map_single_input_to_list(input[1]),
+    )
+
+
+def _map_single_input_to_list(
+    input: SingleInputType,
+) -> list[str] | None:
+    if input is None:
+        return None
+    elif isinstance(input, str):
+        return [input]
+    else:
+        return input
+
+
+class MetricInputs:
+    """A helper class to handle the inputs for the metric in a consistent way."""
+
+    def __init__(
+        self,
+        single_inputs: dict[str, SingleInputType],
+        pairwise_inputs: dict[str, tuple[SingleInputType, SingleInputType]]
+        | None = None,
+        required_params: list[str] | None = None,
+        optional_params: list[str] | None = None,
+        input_record_mapping: dict[str, str] | None = None,
```

Comment:
If you add an empty list/dict as a default argument, the object is initialized when the function is defined and the same reference will be used through the runtime. This could cause unexpected behaviors, so it's generally not recommended.

Ref: https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m

Reply:
ugh yeah forgot about that... makes sense!